internal/shared/domain.go

```diff
@@ -119,6 +119,42 @@ func (dc *Config) Validate(allowedPaths []string) error {
 	return mErr.ErrorOrNil()
 }
 
+func (dc *Config) Copy() *Config {
+	copy := &Config{
```

Comment:
In consul we usually do hand-rolled copies by something like:
```
// shallow copy all non-pointer, non-reference fields (bool, string, uint) by dereferencing the pointer on the stack
dup := *dc
// shallow clone all simple slices
dup.CMDs = slices.Clone(dc.CMDs)
...(etc)...
// recursively call other hand-rolled methods for other things, like Timezone and Osvariant
...(diy)...
// return a pointer to the copy
return &dup
```
The only practical difference between this and what you have is that you are not actually copying the slices, but copying a reference to the top of the slice, so mutating either copy bleeds over into the parent, so you may want some `slices.Clone` calls in here.

Reply:
Let's future proof this and change it